### PR TITLE
support drm exoplayer

### PR DESCRIPTION
**Pull request by yaraht17**
I add new props for support drm for exoplayer-android 
- drmName : enum('widevine', 'playready', 'cenc') 
- drmUrl :  string
- drmHeader: Object. Example: {
                    'Authorization' : `Bearer ${drmKey}` 
                  }
It's working with me. Hope help everyone!

### CODE REVIEW

**Comment by yaraht17** [comment]
check it in repo https://github.com/yaraht17/react-native-video. Support drm, update exo 2.5.0 and support rotate transform view

**Comment by nampdn** [comment]
This is totally awesome, can someone take a look at this?

**Comment by acollazomayer** [comment]
Can you give an example on how to use this?? this is awesome if i could make it work

**Comment by yaraht17** [comment]
@acollazomayer ac
 @all 
Open your node_module and change file like me. After that, call Video component with drm props. Run and pray :))
```
<Video
          ref={videoPlayer => (this.player.ref = videoPlayer)}
          resizeMode={this.state.resizeMode}
          volume={this.state.volume}
          paused={this.state.paused}
          muted={this.state.muted}
          rate={this.state.rate}
       //drm is here 
          drmName={this.props.drmName}
          drmHeader={this.props.drmHeader}
          drmUrl={this.props.drmUrl}
          {...this.props}
          onLoadStart={this.events.onLoadStart}
          onProgress={this.events.onProgress}
          onError={this.events.onError}
          onLoad={this.events.onLoad}
          onEnd={this.events.onEnd}
          onBuffer={this.events.onBuffer}
          style={[styles.player.video, this.styles.videoStyle]}
          repeat={false}
          playWhenInactive={true}
          source={this.props.source} />
```

**Comment by acollazomayer** [comment]
@yaraht17 thanks i will try it for sure! does this work with ios?? If not Is there any chance to make it work?

**Comment by yaraht17** [comment]
@acollazomayer  it not support for ios, only support for android with exoplayer. You should find DRM solution for ios and integration in native module be like this branch. Happy coding!

**Comment by cobarx** [comment]
@yaraht17 Thank you for putting this together. I'd love to get this merged. I've left you some review notes and also the merge conflicts need to get resolved since a lot has changed and the project has been updated to ExoPlayer 2.7.3.

I thoroughly test all PRs before merging, so I need a sample URL and DRM info to be able to put this through its paces. Any chance you can provide these?

Edit: fix typos

**Comment by yaraht17** [comment]
Hmm, And now, i can't provider sample url and drm , because i change my job, i not working with video drm any more. Hope someone working with it and can be provide it for testing. 
@lehoi2195 Can you help us?

**Comment by acollazomayer** [comment]
@yaraht17 hi yaraht i can help you with that. However I can not give you the links I can test it locally

**Comment by acollazomayer** [comment]
@cobarx hey, i can hep you with the testing and also dash if player provides some url for livestremems with drm to test 
https://github.com/Dash-Industry-Forum/dash.js/tree/development/samples/drm

**Comment by acollazomayer** [comment]
@cobarx if youn need anything to get this merged i can help with it.

**Comment by cobarx** [comment]
@acollazomayer Thanks for those samples, that will help a lot!

Can you see about getting the code updated to use the latest version from ExoPlayer 2.8.2 and making sure it works. I can test it using the info you sent over. Otherwise it will be a bit before I get to this.

**Comment by acollazomayer** [comment]
@cobarx yeah sure. I will get right on it

**Comment by brentluna** [comment]
@acollazomayer Thanks for the update. So does the PR not work successfully on android?

**Comment by acollazomayer** [comment]
@brentluna it should work. I did not had the time to test it properly

**Comment by acollazomayer** [comment]
@cobarx Hi cobarx. I made the update of the code. However, it's not working fine. The license server is not getting called sometimes randomly and I have to change the uri two times to get it reproducing. 

Trying to fix these errors. However my Java is limited

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 1]
```diff
@@ -632,4 +696,60 @@ public void setPlayInBackground(boolean playInBackground) {
...
+     */
+    public void setDrmName(String drmName) throws ParserException{
+        switch (Util.toLowerInvariant(drmName)) {
+            case "widevine":
+                this.drmUUID = C.WIDEVINE_UUID;
+                break;
+            case "playready":
+                this.drmUUID = C.PLAYREADY_UUID;
+                break;
+            case "cenc":
```
Comment (danielmarino24i) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:717]: I am not a DRM expert but AFAIK cenc != clearkey.
Clearkey is a protection system like Widevine / PlayReady / Verimatrix / etc

And CENC is common encryption, and is an encryption format. AFAIK Clearkey / PlayReady / Widevine can be encrypted in CENC

https://www.html5rocks.com/en/tutorials/eme/basics/#common-encryption

**Review by Nehru-Wairokpam — APPROVED** [review]

**Comment by atha-cyanberg** [comment]
Even in android it is not working!

**Pull request closed without merging**